htdocs/don/list.php

```diff
@@ -219,9 +229,20 @@
 		$donationstatic->ref=$objp->rowid;
 		$donationstatic->lastname=$objp->lastname;
 		$donationstatic->firstname=$objp->firstname;
-		print "<td>".$donationstatic->getNomUrl(1)."</td>\n";
-        print "<td>".$objp->societe."</td>\n";
-		print "<td>".$donationstatic->getFullName($langs)."</td>\n";
+		print "<td>".$donationstatic->getNomUrl(1)."</td>";
+    if (! empty($conf->global->DONATION_USE_THIRDPARTIES)) {
+    
+    $company=new Societe($db);
+    $result=$company->fetch($objp->socid);
+    if  (!empty($objp->socid) && $company->id > 0)  {
+    print "<td>".$company->getNomUrl(1)."</td>";
+    } else {
+    print "<td>".$objp->societe." ".$donationstatic->getFullName($langs)."</td>";
+    }
+    } else {
+    print "<td>".$objp->societe."</td>";
+		print "<td>".$donationstatic->getFullName($langs)."</td>";
```

Comment:
The name of the donator/contact can be kept event when option DONATION_USE_THIRDPARTIES is on ?
The only change should be to use the key instead of a free text, shouldn't it ?

Reply:
@eldy i keep donator name just replace company / thirdparty. Is it ok ?